### PR TITLE
Restrict access if all gate headers are present

### PR DESCRIPTION
**Pull request by isaacm**

### CODE REVIEW

File: eom/bastion.py [hunk 1]
```diff
@@ -57,9 +57,19 @@
...
-                default=[])
+    cfg.ListOpt(
+        'unrestricted_routes',
+        help='List of paths to allow through the gate.',
+        default=[]
+    ),
+    cfg.ListOpt(
+        'gate_headers',
+        help=(
+            "List of wsgi 'HTTP_' headers. If all of the headers are present, "
```
Comment (TheSriram) [inline eom/bastion.py]: "If all of the headers are present" -> "If all of the headers are <b>not</b> present"

Comment (isaacm) [inline eom/bastion.py]: At the moment, the presence of all `gate_headers` in an incoming request yields 404 error. Would the logic below need a change if I change the description here?

Comment (TheSriram) [inline eom/bastion.py]: Yeah you're right. I didnt read it properly.

No changes required.

**Comment by TheSriram** [comment]
:+1:

File: eom/bastion.py (hunk 1 above)
Comment (BenjamenMeyer) [inline eom/bastion.py]: I know this has already been merged; but I had a question...

Q: Is this an ```all``` or ```any``` match? Line 115 ( https://github.com/racker/eom/pull/84/files#diff-9f08b8496e134c5ec30919d12cd8253eR115 ) looks more like an ```any``` match.

Comment (TheSriram) [inline eom/bastion.py]:
```python
In [1]: gate_headers = ['X-Forwarded-For','MagicHeader']

In [2]: env = {'X-Forwarded-For','MagicHeader','NotsoMagic'}

In [3]: contains_gate_headers = set(gate_headers).issubset(set(env))

In [4]: contains_gate_headers
Out[4]: True

In [5]: env = {'X-Forwarded-For','NotsoMagic'}

In [6]: contains_gate_headers = set(gate_headers).issubset(set(env))

In [7]: contains_gate_headers
Out[7]: False
```

**Pull request merged**